docs/scripting-ref.rst

```diff
@@ -346,30 +346,47 @@ a table to an Array use :ref:`treat-as-array`.
 
 JavaScript → Lua conversion rules:
 
-==============  =================
-JavaScript      Lua
-==============  =================
-string          string
-number          number
-boolean         boolean
-Object          table
-Array           table, marked as array (see :ref:`treat-as-array`)
-``undefined``   ``nil``
-``null``        ``""`` (an empty string)
-Date            string: date's ISO8601 representation, e.g. ``1958-05-21T10:12:00Z``
-function        an empty table ``{}`` (don't rely on it)
-==============  =================
-
-Function arguments and return values are passed by value. For example,
-if you modify an argument from inside a JavaScript function then the caller
-Lua code won't see the changes, and if you return a global JS object and modify
-it in Lua then object won't be changed in webpage context.
+================  =================
+JavaScript        Lua
+================  =================
+string            string
+number            number
+boolean           boolean
+Object            table
+Array             table, marked as array (see :ref:`treat-as-array`)
+``undefined``     ``nil``
+``null``          ``""`` (an empty string)
+Date              string: date's ISO8601 representation, e.g.
+                  ``1958-05-21T10:12:00.000Z``
+function          ``nil``
+circular object   ``nil``
+host object       ``nil``
+================  =================
 
 .. note::
 
     The rule of thumb: if an argument or a return value can be serialized
     via JSON, then it is fine.
 
+Note that currently you can't return DOM Elements, JQuery $ results and
+similar structures from JavaScript to Lua; to pass data you have to
+extract their attributes of interest as plain strings/numbers/objects/arrays:
```

Comment:
so we can't return circular objects and functions from js calls now, is this the only backward incompatible change added by this PR?